File: bindings/java/src/main/java/systems/zlink/contracts/service/spot/ServiceEnums/SpotKind.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.contracts.service.spot;

/** The kind of a spot. */
public enum SpotKind {
    /** No spot kind (unset). */
    INVALID,
    /** A node's well-known entry spot. */
    ENTRY,
    /** A user-created spot. */
    USER
}

File: framework/languages/node/packages/framework/src/contracts/Spots/SpotKind.ts
export enum ZLinkSpotKind {
  Invalid = 'invalid',
  Entry = 'entry',
  User = 'user'
}

export function zlinkSpotKindToWire(kind: ZLinkSpotKind): number {
  switch (kind) {
    case ZLinkSpotKind.Entry: return 1;
    case ZLinkSpotKind.User: return 2;
    default: return 0;
  }
}

export function zlinkSpotKindFromWire(value: number): ZLinkSpotKind {
  switch (value) {
    case 1: return ZLinkSpotKind.Entry;
    case 2: return ZLinkSpotKind.User;
    default: return ZLinkSpotKind.Invalid;
  }
}
